plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/maven2/metadata/Maven2Metadata.java

```diff
@@ -0,0 +1,390 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.maven.internal.maven2.metadata;
+
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import org.joda.time.DateTime;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Maven 2 metadata VO.
+ *
+ * @since 3.0
+ */
+public class Maven2Metadata
+{
+  public enum Level
+  {
+    GROUP, ARTIFACT, BASEVERSION;
+  }
+
+  // G level
+
+  public static class Plugin
+  {
+    private final String artifactId;
+
+    private final String prefix;
+
+    private final String name;
+
+    private Plugin(final String artifactId, final String prefix, final String name) {
+      this.artifactId = artifactId;
```

Comment:
All these classes have private ctors and are called from static helper methods that apply CNNs and other validation.